Core/Services/CarService.cs

```diff
@@ -38,41 +38,47 @@ public class CarService : ICarService
         public async Task<CarDTO> AddCarAsync(CreateCarDTO createCarDTO, string userId)
         {
             var car = _mapper.Map<Car>(createCarDTO);
-            if (IsCarExist(car))
+            if (await IsCarExist(car))
             {
                 throw new HttpException(ErrorMessages.AddingCarNotAllowed, HttpStatusCode.NotAcceptable);
             }
 
             car.CreationDate = DateTimeOffset.UtcNow;
-            car.Category = _categoryRepository.Query().First(c => c.Name == createCarDTO.CategoryName);
+            car.Category = await _categoryRepository.GetBySpecAsync(new CarCategorySpecification.GetByName(createCarDTO.CategoryName));
 
             var user = await _userManager.FindByIdAsync(userId);
             ExceptionMethods.UserNullCheck(user);
             car.User = user;
             car.User.HasCar = true;
 
-            car = await _carRepository.InsertAsync(car);
-            await _carRepository.SaveChangesAsync();
+            car = await _carRepository.AddAsync(car);
             return _mapper.Map<CarDTO>(car);
         }
 
         public IQueryable<CarDTO> GetAllUserCars(string userId)
```

Comment:
This method doesn't load the data from db into memory because of IQueryable storing data out-of-memory. Is it OK?